decimal64scan.go

```diff
@@ -56,11 +55,15 @@ func (d *Decimal64) Scan(state fmt.ScanState, verb rune) error {
 			*d = NegInfinity64
 		}
 		return nil
-	case "nan", "NaN":
-		*d = QNaN64
+	case "nan", "NaN", "qNaN", "qnan", "Nan", "qNan":
```

Comment:
if the `strings.ToLower()` use above is viable then this can become a bit less longform:

```go
case "nan", "qnan":
```

Reply:
but if this isn't appropriate, then probably worth keeping all variants grouped together (all variants of "NaN" and then all variants of "qNaN") just for scannability

Reply:
Do like this idea, Pythons implementation seems to follow this behaviour